src/databricks/labs/ucx/hive_metastore/table_migrate.py

```diff
@@ -503,3 +510,174 @@ def _sql_alter_from(self, table: Table, target_table_key: str, ws_id: int):
     def _is_migrated(self, schema: str, table: str) -> bool:
         index = self._migration_status_refresher.index()
         return index.is_migrated(schema, table)
+
+
+class ACLMigrator:
+    def __init__(
+        self,
+        tables_crawler: TablesCrawler,
+        grant_crawler: GrantsCrawler,
+        workspace_info: WorkspaceInfo,
+        backend: SqlBackend,
+        group_manager: GroupManager,
+        migration_status_refresher: MigrationStatusRefresher,
+        principal_acl: PrincipalACL,
+    ):
+        self._tc = tables_crawler
+        self._gc = grant_crawler
+        self._backend = backend
+        self._workspace_info = workspace_info
+        self._group = group_manager
+        self._migration_status_refresher = migration_status_refresher
+        self._principal_acl = principal_acl
+
+    def migrate_acls(
+        self,
+        *,
+        target_catalog: str | None = None,
+        legacy_table_acl: bool = True,
+        principal: bool = True,
+        hms_fed: bool = False,
+    ) -> None:
+        acl_strategies = []
+        workspace_name = self._workspace_info.current()
+        if legacy_table_acl:
+            acl_strategies.append(AclMigrationWhat.LEGACY_TACL)
+        if principal:
+            acl_strategies.append(AclMigrationWhat.PRINCIPAL)
+        all_grants_to_migrate = self._gc.snapshot()
+        all_migrated_groups = self._group.snapshot()
+        all_principal_grants = self._principal_acl.get_interactive_cluster_grants()
+        tables = self._tc.snapshot()
+
+        if not tables:
+            logger.info("No tables found to acl")
+            return
+        if hms_fed:
+            tables_to_migrate = self._get_hms_fed_tables(tables, target_catalog if target_catalog else workspace_name)
+        else:
+            tables_to_migrate = self._get_migrated_tables(tables)
```

Comment:
```suggestion
            tables_to_migrate = self._get_migrated_tables(tables, target_catalog)
```
BUG: did you forget to pass the target catalog argument to this method?